### DISCUSSION TITLE
Questions RE: latest firmware

### DISCUSSION THREAD

**Discussion by gohills** [thehighnibble/thehighnibble.github.io#78] [IMSAI8080esp replica]
I upgraded from 9 to 10 to 12 without apparent drama.  The GUI UI is up and running and I believe I have managed to boot CPM but my only confirmation is the somewhat familiar pattern of lites because as previously posted I have totally failed at establishing a serial console connection.  I can live without the serial connection when things are working BUT now I the GUI  TTY does not respond.  Did I forget or miss a step in the process to get the TTY connected?  config info  (via the GUI) is below.  

### Network configuration
NTP_SERVER=pool.ntp.org
TZ=AEST-10AEDT,M10.1.0,M4.1.0
HOSTNAME=imsai8080
PORT=80
SSID=mySSID
PASSWORD=myPASSWORD
WIFI.sta.scan=1
#WIFI.password.hide=1
### UART configuration
#UART0=115200,cs8
#UART1=115200,cs8
### Performance parameters
#TTY.netsrv.buffer_delay=33
#LPT.netsrv.buffer_delay=33
### Harddisk image
HARDDISK=hd-ws4
### SIO-2 default port mappings (HAL)
#SIO1.portA.device=WEBTTY,UART0
#SIO1.portB.device=VIOKBD
#SIO2.portA.device=WEBPTR,UART1
#SIO2.portB.device=MODEM
### Modem initialization string. Example:
#MODEM.init=ATS0=1S15=1&A1 #S0=1 auto-answer after 1 ring, S15=1 enable telnet protocol
MODEM.init=ATS15=1 #S15=1 enable telnet protocol

# SIO 1 Channel A, Ports 2/3 connected to terminal (default)
sio1a_upper_case	0
sio1a_strip_parity	0
sio1a_drop_nulls	1
sio1a_baud_rate		115200 
# typical baud rate values are 110, 300, 1200, 2400, 4800, 9600,
# 19200, 38400, 57600, 115200 - 0 = unlimited baud rate

# SIO 1 Channel B, Ports 4/5 connected to keyboard
sio1b_upper_case	0
sio1b_strip_parity	0
sio1b_drop_nulls	0
sio1b_baud_rate		1200 

# SIO 2 Channel A, Ports 34/35 connected to UART1/RS232-2 (default)
# use 8bit here, so that serial file transfers work
sio2a_upper_case	0
sio2a_strip_parity	0
sio2a_drop_nulls	0
sio2a_baud_rate		9600 
# typical baud rate values are 110, 300, 1200, 2400, 4800, 9600,
# 19200, 38400, 57600, 115200 - 0 = unlimited baud rate

# SIO 2 Channel B, ports 36/37 connected to 'AT' telnet modem (default)
# use 8bit here, so that serial file transfers work
sio2b_upper_case	0
sio2b_strip_parity	0
sio2b_drop_nulls	0
sio2b_baud_rate		2400 

# front panel framerate
fp_fps			60
# front panel horizontal size in pixels
fp_size			800
# front panel port value for machine without fp in hex (00 - FF)
fp_port			0

# VIO background and foreground colors in hex RGB format
# white Monitor
vio_bg			303030
vio_fg			FFFFFF
# green Monitor
#vio_bg			002000
#vio_fg			00BF00
# amber Monitor
#vio_bg			202000
#vio_fg			FFBF00
# add scanlines to VIO monitor, 0 = no scanlines
vio_scanlines		1

# <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
# memory configurations in pages a 256 bytes
#       start,size (numbers in decimal, hexadecimal, octal)
# up to 7 memory sections allowed
# up to 6 ram/rom statements per section allowed
# <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

[MEMORY 1]
# Default memory configuration used in most situations with MPU-A ROM:
# 256 pages RAM, 8 pages ROM (replaces or overlays RAM)
ram         0,256
rom         0xd8,8,mpu-a-rom.hex
# Start address of the boot ROM
boot        0xd800

[MEMORY 2]
# Memory configuration for running MPU-A ROM and VIO ROM:
# 240 pages RAM, 8 pages ROM (replaces or overlays RAM)
# 8 pages of RAM and 8 pages of ROM for the VIO from 0xf000
ram         0,0xf0
rom         0xd8,8,mpu-a-rom.hex
ram         0xf0,8
rom         0xf8,8,viofm1.hex
# Start address of the boot ROM
boot        0xd800

[MEMORY 3]
# Memory configuration for running with MEMON80 ROM:
# 248 pages RAM, 8 pages ROM
ram         0,0xf8
rom         0xf8,8,memon80.hex
# Start address of the boot ROM
boot        0xf800

[MEMORY 4]
# Memory configuration for running ROM based XYBASIC:
# 16K ROM, 48K RAM
rom         0,64,xybasic.hex
ram         64,192
# Start address of the boot ROM
boot        0x0000

**Comment by GrantMeStrength** [comment]
can you even ping the web server IP address?

**Reply by gohills** [reply]
Oh yes!  The GUI launches just fine.   When I start the TTY a window pops up but no output to the screen and no input from keyboard is echoed.  I'm starting to wonder if cpm is actually running.  With the new method of storing rom files in a folder do I still need rom statements in the boot configuration?  I'm thinking it's something basic like that

**Comment by oilcan-productions** [comment]
Is the plug icon on the TTY window green or white or red?  If it is white or red pop open the CPA control panel in the UI and tap the Run switch or press RUN on the front panel. Once the icon is green, hit enter a couple of times and it should boot provided a CPM floppy is in the A: drive and you have toggled all the right switches as outlined in the manual :)

**Comment by thisspaceleftblank** [comment]
Did you check the CRT: window?
